src/category_theory/sites/sheaf.lean

```diff
@@ -0,0 +1,830 @@
+  (t : is_sheaf_for P R) {x : family_of_elements P R} (hx : x.compatible) (f : Y ⟶ X) (Hf : R f) :
+  P.map f.op (t.amalgamate x hx) = x f Hf :=
+t.is_amalgamation hx f Hf
+
+/-- If `P` is a sheaf for `R`, it is separated for `R`. -/
+lemma is_sheaf_for.is_separated_for : is_sheaf_for P R → is_separated_for P R :=
+λ q, (is_separated_for_and_exists_is_amalgamation_iff_sheaf_for.2 q).1
+
+/-- C2.1.3 in [Elephant] -/
+lemma is_sheaf_for_iff_generate :
+  is_sheaf_for P R ↔ is_sheaf_for P (generate R) :=
+begin
+  rw ← is_separated_for_and_exists_is_amalgamation_iff_sheaf_for,
+  rw ← is_separated_for_and_exists_is_amalgamation_iff_sheaf_for,
+  rw ← is_separated_for_iff_generate,
+  apply and_congr (iff.refl _),
+  split,
+  { intros q x hx,
+    apply exists_imp_exists _ (q _ (hx.restrict (le_generate R))),
+    intros t ht,
+    simpa [hx] using is_amalgamation_extend _ _ ht },
+  { intros q x hx,
+    apply exists_imp_exists _ (q _ (hx.sieve_extend _)),
+    intros t ht,
+    simpa [hx] using is_amalgamation_restrict (le_generate R) _ _ ht },
+end
+
+/--
+Every presheaf is a sheaf for the family {𝟙 X}.
+
+[Elephant] C2.1.5(i)
+-/
+lemma is_sheaf_for_singleton_iso (P : Cᵒᵖ ⥤ Type v) :
+  is_sheaf_for P (presieve.singleton (𝟙 X)) :=
+begin
+  intros x hx,
+  refine ⟨x _ (presieve.singleton_self _), _, _⟩,
+  { rintro _ _ ⟨rfl, rfl⟩,
+    simp },
+  { intros t ht,
+    simpa using ht _ (presieve.singleton_self _) }
+end
+
+/--
+Every presheaf is a sheaf for the maximal sieve.
+
+[Elephant] C2.1.5(ii)
+-/
+lemma is_sheaf_for_top_sieve (P : Cᵒᵖ ⥤ Type v) :
+  is_sheaf_for P ((⊤ : sieve X) : presieve X) :=
+begin
+  rw ← generate_of_singleton_split_epi (𝟙 X),
+  rw ← is_sheaf_for_iff_generate,
+  apply is_sheaf_for_singleton_iso,
+end
+
+/--
+If `P` is a sheaf for `S`, and it is iso to `P'`, then `P'` is a sheaf for `S`. This shows that
+"being a sheaf for a presieve" is a mathematical or hygenic property.
+-/
+lemma is_sheaf_for_iso {P' : Cᵒᵖ ⥤ Type v} (i : P ≅ P') : is_sheaf_for P R → is_sheaf_for P' R :=
+begin
+  rw [is_sheaf_for_iff_generate, yoneda_condition_iff_sheaf_condition, is_sheaf_for_iff_generate,
+      yoneda_condition_iff_sheaf_condition],
+  intros h f,
+  obtain ⟨g, hg₁, hg₂⟩ := h (f ≫ i.inv),
+  refine ⟨g ≫ i.hom, by simpa [iso.eq_comp_inv] using hg₁, _⟩,
+  { intros g' hg',
+    rw ← iso.comp_inv_eq,
+    apply hg₂,
+    rw reassoc_of hg' },
+end
+
+/--
+If a family of arrows `R` on `X` has a subsieve `S` such that:
+
+* `P` is a sheaf for `S`.
+* For every `f` in `R`, `P` is separated for the pullback of `S` along `f`,
+
+then `P` is a sheaf for `R`.
+
+This is closely related to [Elephant] C2.1.6(i).
+-/
+lemma is_sheaf_for_subsieve_aux (P : Cᵒᵖ ⥤ Type v) {S : sieve X} {R : presieve X}
+  (h : (S : presieve X) ≤ R)
+  (hS : is_sheaf_for P S)
+  (trans : ∀ ⦃Y⦄ ⦃f : Y ⟶ X⦄, R f → is_separated_for P (S.pullback f)) :
+  is_sheaf_for P R :=
+begin
+  rw ← is_separated_for_and_exists_is_amalgamation_iff_sheaf_for,
+  refine ⟨_, _⟩,
```

Comment:
This is conventionally called `split`, right? :-)

Reply:
If you insist :P 
Thanks!